### PR TITLE
Task11 user delete own comment

### PR DESCRIPTION
**Pull request by bluluke**

### CODE REVIEW

File: src/components/CommentCard.jsx [hunk 1]
```diff
@@ -1,11 +1,57 @@
...
+        });
+    };
+    const handleDeleteClick = (event) => {
+        setIsClicked(true);
+     }
+    const optimisticDeleteComment = (commentIdToDelete) => {
+        console.log('first part');
+        setComments((currentComments) => 
+            currentComments.filter((comment) => comment.comment_id !== commentIdToDelete)
+        );
```
Comment (robert112233) [inline src/components/CommentCard.jsx:42]: I would be hesitant to delete _this_ optimistically. Deleting usually involves sensitive data and we wouldn't want a scenario where someone quickly deletes a controversial comment, sees it disappear then instantly logs off. The popup error message is great but we do have to _wait_ for the request to fail in order to see it. If we were using slow 3G this error message might be too late. 

There is no right/wrong for optimistic rendering these are just my thoughts. It's important you have reasoning for using it/not using it. 

I would consider placing optimisticDeleteComment() inside the .then block. That way you don't have to refresh to see it deleted, but you _do_ have to ensure the request has been fully successful first 👍

File: src/components/CommentCard.jsx [hunk 2]
```diff
@@ -1,11 +1,57 @@
...
+        toast("There was a problem. Comment was not deleted.", {
+            position: "top-center",
+            bodyClassName: 'toastBody',
+        });
+    };
+    const handleDeleteClick = (event) => {
+        setIsClicked(true);
+     }
+    const optimisticDeleteComment = (commentIdToDelete) => {
+        console.log('first part');
```
Comment (robert112233) [inline src/components/CommentCard.jsx:39]: Remove console.log()

**Review by robert112233 — APPROVED** [review]
Great work Luke! Please see my feedback below, I'll send you a screenshot on slack of some other errors, I'm unsure why I'm getting them. 

Fix the errors and think about your optimistic rendering, it's up to you if you want to use it or not 👍

File: src/components/CommentCard.jsx [hunk 3]
```diff
@@ -1,11 +1,55 @@
...
+    useEffect(() => {
+        if(isClicked) {
+        deleteComment(commentIdToDelete).then(() => {
+            optimisticDeleteComment(commentIdToDelete) 
+        })
+        .catch((err) => {
+            handleError()   
+        })
+        }
+    }, [isClicked])
```
Comment (popatre) [inline src/components/CommentCard.jsx:25]: Does this need a use effect?

Currently, you're clicking a button, to change to state, the state change is then triggering a use effect.

Could you think of a refactor where you can click the button, and it invokes the function? - You already have the function that needs to be invoked - there no real need for the useEffect.

**Review by popatre — CHANGES_REQUESTED** [review]
- Great work making it so that only a logged in user can delete comments
- Nice feedback/updating of the UI when a comment gets deleted. 
- Nice handling of the user being able to only delete their own comments
- i like the error toast notification 

- [ ] Would be good if you disabled the delete button after clicking so they can't click it multiple times to delete the same comment - on a slow connection i can keep clicking the button and it looks like nothing it happening. Would be better for the button to be disabled and say deleting..., or something. Just to let us know something going on

- [ ] Please see my comment regarding the useEffect. You can clean this up and get rid of that, theres not really any need for it. You only have a handle delete that you can use to call any function you need, at the moment, you've got it setting state to call the useEffect

**Review by roisinoreilly — APPROVED** [review]
thanks for refactoring, this works great! Merge when ready :)

**Pull request merged**